frontend/test/pytest/test_aws_braket_devices.py

```diff
@@ -0,0 +1,702 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for `OpenQasmDevice` on Amazon Braket devices
+"""
+import numpy as np
+import pennylane as qml
+import pytest
+
+from catalyst import grad, qjit
+
+pytest.importorskip("braket")
+
+
+@pytest.mark.braket
+class TestBraketGates:
+    """Unit tests for quantum gates."""
+
+    @pytest.mark.parametrize(
+        "device",
+        [
+            qml.device(
+                "braket.aws.qubit",
+                device_arn="arn:aws:braket:::device/quantum-simulator/amazon/sv1",
+                wires=3,
```

Comment:
@dime10 @maliasadi once this PR is merge-ready, do we want to modify the actions to selectively _not_ run the tests that run on `braket.aws.qubit`?

We could do this by specifying a 'mark' for this particular device option, and then using `pytest -k "not remoteaws"` or something.

This is because the tests that run on the AWS API will charge us.

Reply:
There are only a few tests running on `braket.aws.qubit` and we can definitely exclude them all from being tested remotely. I'll fix this. Thanks!

Reply:
We should probably always run tests on the local simulator, but not the remote one, that is a good point.

Reply:
Fixed!